test/mock-apis.ts

```diff
@@ -0,0 +1,409 @@
+import WebSocket from 'ws';
```

Comment:
I didn't really like having the mock-apis in the same file as test-utils. 1) for organization reasons and 2) because mock-apis does add things to the window, it will cleaner for unit tests to not have that overhead (and sometimes unexpected behavior).